### PR TITLE
DA-4596: update for auth0 jwks cache changes

### PR DESCRIPTION
**Pull request by ajinkyan83**
Signed-off-by: Ajinkya Nahar <ajinkyan@proximabiz.com>

### CODE REVIEW

File: auth0/token.go [hunk 1]
```diff
@@ -74,7 +75,7 @@ func NewAuth0Client(env,
 }
 
 // GetToken ...
-func (a *ClientProvider) GetToken() (string, error) {
+func (a *ClientProvider) GetToken(input bool) (string, error) {
```
Comment (enkhalifapro) [inline auth0/token.go:78]: the input is useless after caching the cert, no need for it

Comment (ajinkyan83) [reply auth0/token.go]: done!

**Review by lukaszgryglicki — APPROVED** [review]
/lgtm

File: auth0/token.go [hunk 2]
```diff
@@ -349,22 +317,28 @@ func (a *ClientProvider) RefreshToken() (RefreshResult, error) {
...
+		_, err = a.refreshCachedToken()
 		if err != nil {
 			return RefreshError, err
 		}
+
 		return RefreshSuccessful, nil
 	}
 
-	ok, claims, err := a.isValid(authToken)
+	ok, claims, err := a.isValid(authToken, false)
```
Comment (khalifapro) [inline auth0/token.go:328]: No need to the second input

Comment (ajinkyan83) [reply auth0/token.go:328]: I think it is good to be precise by specifying the second input.

**Pull request merged**